### DISCUSSION TITLE
Question regarding uwhoisid

### DISCUSSION THREAD

**Discussion by swiss-JF** [CIRCL/lookyloo#776] [General]
Hello,

I spent my whole afternoon trying to make it work, but none of my attempts was successful. It is probably something small, but I am keeping missing it.

I wanted to install the uwhoisid, but I did not make it to work.

I followed strictly the instruction on here: [https://www.lookyloo.eu/docs/main/install-lookyloo.html](url), but I keep getting error messages and uwhoisid didn't work.
![image](https://github.com/Lookyloo/lookyloo/assets/63421039/b21c973d-3387-4713-a28d-e879eeb28344)

My guess is, something is wrong with the uwhoisd.service.sample file. I put the following values in there. Not sure if I should put a ~before the / or if I should put the WorkingDirectory within ""
![image](https://github.com/Lookyloo/lookyloo/assets/63421039/ae8d5d00-c2f9-4f48-9115-a8e645597667)

However, I am unsure it is this, as  the error message mentions a failure "to determine user credentials: No such process"
I have it installed in a Linux Mint virtual machine running on VirtualBox named: mint@mint-VirtualBox

Any ideas welcome, thank you!

Ps. I searched for uwhoisd, but did not find a previous discussion on this.

**Comment by Rafiot** [comment]
Okay that seems to fail somewhere with systemd, let's try without it at first.

From the doc, I'm assuming you went to that point: 
```
git clone https://github.com/Lookyloo/uwhoisd
cd uwhoisd
poetry install
```
and then went on with systemd.

What happen if you do `poetry run start` from the `uwhoisd` directory?

If there is no error, try in another terminal on the same machine `telnet localhost 4243` and if you get a telnet prompt, type a domain. If that works, you have a working uwhoisd and it is purely systemd's fault. We can investigate further.

One thing that might be an issue is that you put the `WorkingDirectory` to `/home/mint/opt/lookyloo` when it should be `/home/mint/opt/uwhoisd` if you're attempting to run uwhoisd.

**Comment by swiss-JF** [comment]
Thank you!

In the meantime I had deleted the uwhoisd, so I reinstalled it in `/home/mint/opt/uwhoisd`. There was  an error after `poetry run start`, I put what the system wanted `export UWHOISD_HOME='/home/mint/opt/uwhoisd`.

Afterwards, I opened a new terminal and entered `telnet localhost 4243` and then inputting a random website. So, it seems `uwhoisd` is working. I tried a few stuffs, but what I gut is a "No file/folder found for package lookyloo", despite being in the folder itself and having files there.
<img width="952" alt="image" src="https://github.com/Lookyloo/lookyloo/assets/63421039/df88c067-7e4b-4f3d-bd28-48d52b55c2c5">

However, I went back to `lookyloo`, and I cannot access its web interface or run it in any way
<img width="445" alt="image" src="https://github.com/Lookyloo/lookyloo/assets/63421039/bfc9b7bf-1d4c-4eb1-bef0-a596df96789f">

My guess is that I am not a good poet, and it is probably an issue with poetry. I am a bit in a rush, so will have to get back to this again, but I looked in the `poetry` command and ran a `poetry show` (no `lookyloo` in the list) and `poetry update`. Maybe I corrupted something when I deleted the `uwhoisd` folder. 

No clue where to look for lookylook... ;-)

**Reply by Rafiot** [reply]
Good that uwhoisd is sorted out. `uwhoisd` and `lookyloo` are two independent projects, so they shouldn't impact each other. 

My guess for lookyloo is that you didn't run `poetry install` from the `lookyloo` repository

**Reply by swiss-JF** [reply]
Ok, I was thinking `poetry` could impact the projects. I reinstalled `lookyloo`and reinitialized the configuration files. ´Lookyloo` works, `uwhoisd` works, but i am not sure if they are working together when I do a capture in `lookyloo`.

Based on the integration documentation,  everything seems to be correctly set. Unless the IP address and port needs to be amended. The `auto_trigger_modules` in `config/generic.json` is set to `true`.
![image](https://github.com/Lookyloo/lookyloo/assets/63421039/9cd1bcbf-80ea-4853-a2d5-5389960712f7)
However, I cannot find where the `whois` information shows up, whether on the web interface, nor in `~/lookyloo/scraped/[...]/[Capture_Timestamp]`

Could it be that the `uwhoisd.service` is required and needs to be enabled?

**Reply by Rafiot** [reply]
The IP address only needs to be amended if `uwhoisd` is running from another machine/VM than the one `lookyloo` is running on (if `telnet 127.0.0.1 4243` works from the machine you have lookyloo installed, all good). It should be the case after you ran `poetry run start` from the `uwhoisd` directory. The systemd script is doing the same as `poetry run start`, but makes it reboot-safe.

The whois information should be visible after you triggered a capture in lookyloo. From the tree, when you click on one of the node it will open a popup. In that popup, you should have links to the whois infos:

![image](https://github.com/Lookyloo/lookyloo/assets/248875/63020dea-5969-4463-8439-b3e4e1368bc2)

**Reply by swiss-JF** [reply]
The results I get with the public instance of lookyloo for a random public capture:
<img width="476" alt="image" src="https://github.com/Lookyloo/lookyloo/assets/63421039/8cc9222f-374f-4307-bb78-4c0af389c267">

The results I get with the private instance doesn't have the link to the whois information
<img width="440" alt="image" src="https://github.com/Lookyloo/lookyloo/assets/63421039/8d7920a0-6070-46c1-a8ab-9aa2d34bd78a">


I tried to see if the `uwhoisd` was working, so I repeated the procedure you suggested above and had this:
<img width="776" alt="image" src="https://github.com/Lookyloo/lookyloo/assets/63421039/31581ebf-13a4-451c-aa26-22ec72d0ad6c">

I think I will give it a go in a fresh VM with Ubuntu and see if I get any further on a fresh install.

**Reply by Rafiot** [reply]
That's odd, but it probably means lookyloo cannot connect to uwhoisd. They're both running in the same VM, right? 

And in `config/modules.json` > `UniversalWhois`, `enabled` is `True`? If yes, can you check at the logs? There should be something there with a warning.

**Reply by swiss-JF** [reply]
Yup, it is the same VM

The `config/modules.json` > `UniversalWhois`, are enabled, (set to `true`) [https://www.lookyloo.eu/docs/main/lookyloo-integration.html](integration with 3rd party tools), the `allow_auto_trigger` in `config/generic.json` is set to `true`.

I had some config files twice, one `modules.json` and `modules.json.sample`, the `.sample` ones were set to `false`, I put it to `true`. Then restarted both terminal  for `uwhoisd` and `lookyloo`, once, it didn't work. 

Surprisingly, I repeated the procedure and it works.

The logs file are in `~/lookyloo/config/log`, right? If so they are [https://www.swisstransfer.com/d/d0f53033-3f57-4eb4-9352-931ff33217f4](url) if you want to take a peak at.

It was probably an issue with those settings, but it seems it is solved now.

Thank you for your help!

**Reply by Rafiot** [reply]
The idea for the .sample config files is that if one of the config setting isn't defined in the non-sample file, lookyloo (or the relevant project) will simply pick the value from there. In theory, you shouldn't have to change them and only edit the non-sample file. down the line, it makes it possible for you to run git pull and update the project without conflicts.

when you say "restarted both terminal", do you mean you ran the stop/start commands in each poetry environment? 

Glad it works. And by looking at the logs, it is pretty much all expected, not sure why you had duplicate UUIDs, but it happens when you play around and debug.
